### PR TITLE
Added Prio and PrioID

### PR DESCRIPTION
**Pull request by pscn**
See: https://www.musicpd.org/doc/html/protocol.html

Unfortunately I'm not sure since which protocol version this is supported :(

### CODE REVIEW

File: mpd/client.go [hunk 1]
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
 	return cmd.AttrsList("file")
 }
 
+// Prio set the priority of the specified songs. If end is negative but start
+// is positive, it does it for the song at position start. If both start and
+// end are positive, it does it for positions in range [start, end).
+func (c *Client) Prio(prio, start, end int) error {
```
Comment (fhs) [inline mpd/client.go]: Maybe name this function SetPriority? There is no reason we have to follow MPD's obscure naming.

File: mpd/client.go [hunk 2]
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
...
+		return c.Command("prio %d %d", prio, start).OK()
+	case start < 0 && end >= 0:
+		return errors.New("negative start index")
+	default:
+		panic("unreachable")
+	}
+}
+
+// PrioID set the prio of the song with the given id
+func (c *Client) PrioID(prio, id int) error {
```
Comment (fhs) [inline mpd/client.go]: Name this SetPriorityID?

File: mpd/client.go [hunk 3]
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
 	return cmd.AttrsList("file")
 }
 
+// Prio set the priority of the specified songs. If end is negative but start
+// is positive, it does it for the song at position start. If both start and
```
Comment (fhs) [inline mpd/client.go]: Non-actionable comment: >= 0 mean non-negative, not "positive". I realize the same wording is used in other functions, so don't worry about it.

**Comment by pscn** [comment]
Renaming done.  I missed the part where you do it for volume like this.  I hope I come around to the unit tests later today.

**Comment by pscn** [comment]
@fhs should I add SetPriority and SetPriorityID to the CommandList?

**Comment by fhs** [comment]
> @fhs should I add SetPriority and SetPriorityID to the CommandList?

Sure, if MPD supports it and it works.

**Comment by pscn** [comment]
Added it to CommandList also, as my MPD supports it.  Also fixed an issue with the delete call in CommandList, please let me know if I should do this in a separate pull request.

File: mpd/internal/server/server.go [hunk 4]
```diff
@@ -387,6 +387,66 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
...
+				ack("invalid song position")
+				return
+			}
+			start = i
+			end = i + 1
+		} else if len(rng) == 2 {
+			// Updating a range of the playlist from specified start/end positions.
+			start, err = strconv.Atoi(rng[0])
+			if err != nil {
+				ack("Integer or range expected")
```
Comment (fhs) [inline mpd/internal/server/server.go]: don't captitalize error messages (ditto for errors below)

File: mpd/internal/server/server.go [hunk 5]
```diff
@@ -387,6 +387,66 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
...
+		}
+
+		if len(rng) == 1 {
+			// Updating a single song from the playlist at position i.
+			i, err := strconv.Atoi(rng[0])
+			if err != nil {
+				ack("invalid song position")
+				return
+			}
+			start = i
```
Comment (fhs) [inline mpd/internal/server/server.go]: check `start` is less than `s.currentPlaylist.Len()` and >= 0?

Comment (pscn) [reply mpd/internal/server/server.go]: Checking for start >=0 is sensible, but I don't want to test for s.currentPlaylist.Len() as it doesn't matter for this case.

File: mpd/internal/server/server.go (hunk 4 above)
Comment (pscn) [reply mpd/internal/server/server.go]: That was copied from another case where a range is extracted.  Should I fix those to while I'm at it?

**Comment by pscn** [comment]
I'm failing in using git and my branch is all kinds of messed up :'( Don't really have the time to figure out the correct way to resolve this, so I'm closing this for now and start a clean pull request again later.  Sorry for the waste of time :|

**Pull request closed without merging**